### ISSUE TITLE
Add public API for building Script objects.

### ISSUE THREAD

**Issue by dj2** [google/amber#682]
There have been requests to be able to generate scripts on the fly without writting AmberScript. We should investigate adding something like a ScriptBuilder API into the public API of Amber which lets folks create scripts programatically.

This would depend on #524 so a text script could be generated on failure.

**Comment by dj2** [comment]
@alan-baker @jeffbolznv does this capture the desired functionality?

**Comment by jeffbolznv** [comment]
Right, I'd like to be able to generate an Amber script dynamically inside Vulkan CTS so we can use Amber to execute a bunch of test variants, rather than the one-script-per-file approach that CTS is currently using.
